File: events-notifications/subscriptions/renew-subscription-update-event-filters/code-samples/updateSubscription.js
// https://developers.ringcentral.com/my-account.html#/applications
// Find your credentials at the above url, set them as environment variables, or enter them below

// PATH PARAMETERS
const subscriptionId = '<ENTER VALUE>';

// OPTIONAL QUERY PARAMETERS
const queryParams = {
    //aggregated: true
};

// POST BODY
const body = {
    eventFilters: [
        '<ENTER VALUE>',
    ],
    deliveryMode: {
        transportType: 'PubNub',
        address: '<ENTER VALUE>',
        encryption: true,
        certificateName: '<ENTER VALUE>',
        registrationId: '<ENTER VALUE>',
        verificationToken: '<ENTER VALUE>'
    },
    expiresIn: 604800
};

const SDK = require('ringcentral');
const rcsdk = new SDK({server: process.env.serverURL, appKey: process.env.clientId, appSecret: process.env.clientSecret});
const platform = rcsdk.platform();
platform.login({ username: process.env.username, extension: process.env.extension, password: process.env.password }).then(() => {
    platform.put(`/restapi/v1.0/subscription/${subscriptionId}`, body, queryParams).then((r) => {
        // PROCESS RESPONSE
    });
});

File: events-notifications/subscriptions/renew-subscription-update-event-filters/code-samples/updateSubscription.cs
// https://developers.ringcentral.com/my-account.html#/applications
// Find your credentials at the above url, set them as environment variables, or enter them below

// PATH PARAMETERS
string subscriptionId = "<ENTER VALUE>";

// OPTIONAL QUERY PARAMETERS
UpdateSubscriptionParameters updateSubscriptionParameters = new UpdateSubscriptionParameters {
    //aggregated = true
};

// POST BODY
ModifySubscriptionRequest modifySubscriptionRequest = new ModifySubscriptionRequest {
    eventFilters = new[] {
        "<ENTER VALUE>",
    },
    deliveryMode = new NotificationDeliveryModeRequest {
        transportType = 'PubNub',
        address = "<ENTER VALUE>",
        encryption = true,
        certificateName = "<ENTER VALUE>",
        registrationId = "<ENTER VALUE>",
        verificationToken = "<ENTER VALUE>"
    },
    expiresIn = 604800
};

RestClient rc = new RestClient(
    Environment.GetEnvironmentVariable("clientId"),
    Environment.GetEnvironmentVariable("clientSecret"),
    false
);
await rc.Authorize(
    Environment.GetEnvironmentVariable("username"),
    Environment.GetEnvironmentVariable("extension"),
    Environment.GetEnvironmentVariable("password")
);
var r = await rc.Restapi().Subscription(subscriptionId).Put(modifySubscriptionRequest, updateSubscriptionParameters);
// PROCESS RESPONSE
